### PR TITLE
stop allowing string control_values for ctrl

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
This warning was added eons ago (#2634), but it pre-dates our deprecation management page so it slipped through the cracks.

**Description of the Change:**
Remove the support for passing `control_values` as a string to `qml.ctrl`

**Benefits:**
Deprecation cycle is now complete!

**Possible Drawbacks:**
It wasn't on our deprecations page and was therefore never formally scheduled for removal. But you know.. it's been almost 10 releases.

NOTE: I want to see PennyLaneAI/plugin-test-matrix#57 merged and get a look at the matrix before we merge this into master. Too many things breaking at once 😓

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -46,6 +46,9 @@
   because it depended on the now-deprecated `Observable.return_type` property.
   [(#4762)](https://github.com/PennyLaneAI/pennylane/pull/4762)
 
+* Specifying `control_values` passed to `qml.ctrl` as a string is no longer supported.
+  [(#)]()
```
Comment (timmysilv) [inline doc/releases/changelog-dev.md:50]:
```suggestion
  [(#4816)](https://github.com/PennyLaneAI/pennylane/pull/4816)
```

**Review by mudit2812 — COMMENTED** [review]
Thanks Matt! I've actually always wondered why we decided to do this? `MultiControlledX` uses strings for it's control values, and I'm not very keen on having controlled operators that do not both accept the same input for control values.

**Comment by timmysilv** [comment]
[sc-49926]

**Review by trbromley — APPROVED** [review]
Thanks @timmysilv!

**Review by albi3ro — APPROVED** [review]
Good catch 🎉 Glad we have a more formal deprecation procedure now :)

**Pull request merged**